### PR TITLE
Maintain Buyer's BatPay Balance

### PR DESCRIPTION
**Pull request by nicoayala**
### What does this Pull Request do?
Adds a recurring function that checks Buyer's BatPay balance and issues a deposit transaction when there is not enough balance to make transfers.

### Are there points in the code the reviewer needs to double check?
The module is placed inside a new folder named `recurrent`, I'm open to change it if we do not agree entirely.
**Important:** before sending the deposit BAPI needs to increase the allowance for BatPay. The recurring function first sends the increaseApproval transaction and listens for the `Approval` event. When it arrives, it sends the deposit for the allowed amount.

### Why was this Pull Request needed?
To pay Data Sellers through BatPay, the Buyer must issue a deposit to this contract. If there is not enough balance, transfers will fail.

### Tasks
- [x] Implement SS signature
- [x] Test

### Does this Pull Request meet the acceptance criteria?
- [x] I have read the [Contribution guidelines](https://github.com/wibsonorg/buyer-sdk/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](https://github.com/wibsonorg/buyer-sdk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: buyer-api/src/recurrent/checkBatPayBalance.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ *  It issues a Deposit transaction that will allow the Buyer account to operate
+ *  with the BatPay contract.
+ * @param {ApprovalEvent} event
+ */
+export const sendDeposit = async (event) => {
+  const { owner, spender, value: amount } = event;
+  const { address } = await getAccount();
+  if (address.toLowerCase() !== owner.toLowerCase()) return;
+  if (spender.toLowerCase() !== BatPay.options.address.toLowerCase()) return;
+  await addTransactionJob('Deposit', { amount });
```
Comment (Agus) [inline buyer-api/src/recurrent/checkBatPayBalance.js]: Do we need this `await` 🤔?

Comment (danielfx90) [reply buyer-api/src/recurrent/checkBatPayBalance.js]: yes, if not the log message that follows does not reflect the reality. The key thing is that we are not awaiting the whole function when calling it

**Comment by genuinefafa** [comment]
💰

**Comment by genuinefafa** [comment]
💸

**Comment by danielfx90** [comment]
@manelli please review. If you are OK, merge.

**Pull request merged**